design/marketplace.md

```diff
@@ -162,16 +162,100 @@ The client is able to retrieve any funds that are left in the Request.
     ---------
 
 
+Slot reservations
+-----------------
+
+Competition between hosts to fill slots has some advantages, such as providing
+an incentive for hosts to become proficient in downloading content and
+generating proofs. It also has some drawbacks, for instance it can lead to
+network inefficiencies because multiple hosts do the work of downloading and
+proving, while only one host is rewarded for it. These inefficiencies lead to
+higher costs for hosts, which leads to an overall increase in the price of
+storage on the network. It can also lead to clients inadvertently inviting too
+much network traffic to themselves. Should they for instance post a very
+lucrative storage request, then this invites a lot of hosts to start downloading
+the content from the client simultaneously, not unlike a DDOS attack.
+
+Slot reservations are a means to avoid these inefficiencies. Before downloading
+the content associated with a slot, a host can deposit collateral to reserve the
+slot. When it succeeds in reserving the slot, then no other hosts can fill the
+slot. After the host downloads the content and calculates a proof it moves the
+slot from its reserved state into the filled state. Then it begins to
+periodically provide storage proofs and accrue payments for the slot.
+
+        collateral               proof
+            |                      |
+            v                      v
+            ------------------------------------------------------------------
+     slot:  |/ / / / / / / / / / / |//////////////////////////////////////////
+            ------------------------------------------------------------------
+            |                      |
+            v                      v
+          slot                    slot
+         reserved                filled
+
+
+            ---------------- time ---------------->
+
+
+Slot reservations could however provide a relatively cheap way for malicious
+hosts to sabotage storage requests, by reserving a slot and never filling it. To
+avoid this we burn the collateral of all reserved slots when the request times
+out. Hosts should therefore be careful to only reserve slots for which they are
+confident that they can fill them before the request timeout.
+
+Burning collateral for slots that are in the reserved state when the request
+times out, does however invite a different type of attack. A malicious client
+could cause hosts to lose collateral on purpose by posting a request for storage
+for content that it never intends to release to the network. Once hosts reserve
+slots in this request, they can never fill them because they will be unable to
+provide a storage proof without the content. To avoid this attack we also burn
+some of the funds from the request when a request times out because of slots
+that are in the reserved state. The more of these slots there are, the more the
```

Comment:
Will burning a portion of the client funds have an impact on how much the client would need to fund the request? If there are several withholding/non-filling events but ultimately the request succeeds to its natural conclusion, we want to ensure the request is funded enough to pay the hosts that fulfilled their duties. Perhaps there could be an additional amount required by the client to supply to the request, that would serve as a coffer for such withholding/non-filling events. Every withholding/non-filling event would burn a portion of the funds from the coffer. If there's anything remaining at the conclusion of the contract, it can be withdrawn back to the client. If, however, at some point all of the funds in the coffer are slashed, the request is considered failed.

Reply:
The client funds would only be burned when the request times out. When a request times out, it is in a final state. It cannot start afterwards. If the client wants to try again, they would retrieve the remaining funds from the timed out request, and submit a new request.